src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfStates = [];
+  const copyOfState = { ...state };
+
+  const finiteStateMachine = ({ type, extraData, keysToRemove }) => {
+    switch (type) {
+      case 'addProperties':
+        Object.assign(copyOfState, extraData);
+
+        return { ...copyOfState };
+
+      case 'removeProperties':
+        keysToRemove.forEach(keyToRemove => {
+          delete copyOfState[keyToRemove];
+        });
+
+        return { ...copyOfState };
+
+      case 'clear':
+        Object.keys(copyOfState).forEach(keyToRemove => {
+          delete copyOfState[keyToRemove];
+        });
```

Comment:
we clear all values from our object so can we just assign new empty object to avoid additional loop?